From 0f207a9cc6c0fda8ff47f53598404e1c878f7610 Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@liri.io>
Date: Tue, 20 Mar 2018 18:22:39 +0100
Subject: [PATCH] Remove suffix for Wayland session

Some desktops like GNOME specify which windowing system is in use
with the Name entry of their desktop file.

For Wayland-only desktops such as Liri this information is
redundant and so is for X11-only window managers.

Do not append the Wayland suffix and let desktops handle it
themeselves.

[ChangeLog][Greeter] Remove suffix for Wayland sessions
---
 src/common/Session.cpp | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/src/common/Session.cpp b/src/common/Session.cpp
index 2d7b04f..cb4cef7 100644
--- a/src/common/Session.cpp
+++ b/src/common/Session.cpp
@@ -157,15 +157,8 @@ namespace SDDM {
             if (current_section != QLatin1String("Desktop Entry"))
                 continue; // We are only interested in the "Desktop Entry" section
 
-            if (line.startsWith(QLatin1String("Name="))) {
-                if (type == WaylandSession)
-                    if (line.mid(5).endsWith(QLatin1String(" (Wayland)")))
-                        m_displayName = QObject::tr("%1").arg(line.mid(5));
-                    else
-                        m_displayName = QObject::tr("%1 (Wayland)").arg(line.mid(5));
-                else
-                    m_displayName = line.mid(5);
-            }
+            if (line.startsWith(QLatin1String("Name=")))
+                m_displayName = line.mid(5);
             if (line.startsWith(QLatin1String("Comment=")))
                 m_comment = line.mid(8);
             if (line.startsWith(QLatin1String("Exec=")))
-- 
2.25.1

